internal/pkg/otel/run_test.go

```diff
@@ -0,0 +1,72 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package otel
+
+import (
+	"context"
+	"path/filepath"
+	"sync"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"go.opentelemetry.io/collector/otelcol"
+)
+
+func TestStartCollector(t *testing.T) {
+	testCases := []struct {
+		configFile           string
+		expectedErrorMessage string
+	}{
+		{
+			configFile:           "all-components.yml",
+			expectedErrorMessage: "", // empty string means no error is expected
+		},
+		{
+			configFile:           "nonexistent-component.yml",
+			expectedErrorMessage: `error decoding 'extensions': unknown type: "zpages"`,
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.configFile, func(t *testing.T) {
+			configFiles := []string{"file:" + filepath.Join(".", "testdata", tc.configFile)}
```

Comment:
Surprised that doesn't need to be `file://`

Reply:
Yeah, "file" is the default when there's no scheme provided.

- https://opentelemetry.io/docs/collector/configuration/#location
- https://github.com/open-telemetry/opentelemetry-collector/blob/v0.100.0/confmap/resolver.go#L82-L83
- https://github.com/open-telemetry/opentelemetry-collector/blob/v0.100.0/confmap/resolver.go#L131-L132

Reply:
Oh you mean the `//` in `file://`. Yeah that's how it works.

Reply:
What do you mean that is how it works? It works with `file://` or `file:`? I was just stating that I am surprised if doesn't require the `//` before the actual path.

Reply:
Yeah, by "this is how it works" I meant that this is how it's implemented in the file provider: https://github.com/open-telemetry/opentelemetry-collector/blob/v0.100.0/confmap/provider/fileprovider/provider.go#L42-L73.

Whether it's compliant with the https://en.wikipedia.org/wiki/Uniform_Resource_Identifier#Syntax or https://en.wikipedia.org/wiki/File_URI_scheme or https://datatracker.ietf.org/doc/html/rfc3986 is probably another question. The file provider definitely allows relative paths like `file:./local-file.txt`, which I'm not sure the RFC does.